File: ios/Yacht/ReactNativeBridge/WalletBridge.m
//
//  WalletBridge.m
//  Yacht
//
//  Created by Henry Minden on 10/6/22.
//

#import <Foundation/Foundation.h>
#import "React/RCTBridgeModule.h"
#import "React/RCTEventEmitter.h"

@interface RCT_EXTERN_MODULE(WalletBridge, RCTEventEmitter)

RCT_EXTERN_METHOD(returnLedgerAccount:(NSDictionary*)account)
RCT_EXTERN_METHOD(returnLedgerDevice:(NSDictionary*)device)
RCT_EXTERN_METHOD(returnComponentLoaded)
RCT_EXTERN_METHOD(sendDeviceId)

@end

File: ios/Yacht/ReactNativeBridge/WalletBridge.swift
//
//  WalletBridge.swift
//  Yacht
//
//  Created by Henry Minden on 10/6/22.
//

import Foundation

@objc(WalletBridge)
class WalletBridge: RCTEventEmitter {

    public static var shared: WalletBridge?

    override init() {
        super.init()
        WalletBridge.shared = self
    }

    @objc
    func sendDeviceId(deviceId: String, withAccountIndex: Int) {
        sendEvent(withName: "onSendDeviceId", body: ["deviceId": deviceId, "accountIndex": withAccountIndex])
    }

    @objc
    func returnLedgerAccount(_ account: NSDictionary) {
        NotificationCenter.default.post(name: NSNotification.Name("return.ledger.account"), object: account)
    }
    
    @objc
    func returnLedgerDevice(_ device: NSDictionary) {
        NotificationCenter.default.post(name: NSNotification.Name("return.ledger.device"), object: device)
    }
    
    @objc
    func returnComponentLoaded() {
        NotificationCenter.default.post(name: NSNotification.Name("return.component.loaded"), object: nil)
    }

    override func supportedEvents() -> [String]! {
        return [
                "onSendDeviceId"
                ]
    }
    
    override static func requiresMainQueueSetup() -> Bool {
        return true
    }
    
}
